@utahdts/utah-design-system/react/components/buttons/ConfirmationButton.jsx

```diff
@@ -58,29 +87,95 @@ function ConfirmationButton({
   promptChildren,
   size,
   type,
+  ...rest
 }) {
   const [isClicked, setIsClicked] = useState(false);
-  const buttonClick = () => {
-    setIsClicked(!isClicked);
-    onClick();
-  };
+  const timer = useRef(NaN);
+  const { addAssertiveMessage } = useAriaMessaging();
+
+  const clearTimer = useCallback(() => {
+    if (timer.current) {
+      clearTimeout(timer.current);
+      timer.current = NaN;
+    }
+  }, []);
+
+  const resetButton = useCallback(() => {
+    clearTimer();
+    setIsClicked(false);
+    addAssertiveMessage(children);
+  }, [addAssertiveMessage, children, clearTimer]);
+
+  const handleOnClick = useCallback((e) => {
+    if (!isBusy) {
+      if (isClicked) {
+        onClick?.(e);
+        resetButton();
+      } else {
+        clearTimer();
+        setIsClicked(true);
+        addAssertiveMessage(promptChildren);
+        timer.current = setTimeout(resetButton, 3500);
+      }
+    }
+  }, [addAssertiveMessage, clearTimer, isBusy, isClicked, onClick, promptChildren, resetButton]);
+
+  const renderLeftIcon = useCallback(() => {
```

Comment:
Rendering components should be composable. It would be better to track a flag of which one to render and then do the actual JSX in the return statement. Putting JSX in logic greatly complicates testing and maintainability. Or make it a separate component.

Reply:
Suggestion: Use ternary in the render like:
`
(iconLeft?.initial && !isClicked) ? <span> : null;
(iconLeft?.prompt && isClicked) ? <span> : null;
`